tls/s2n_handshake.c

```diff
@@ -346,3 +346,19 @@ S2N_RESULT s2n_handshake_validate(const struct s2n_handshake *s2n_handshake)
     RESULT_DEBUG_ENSURE(s2n_handshake->message_number >= 0 && s2n_handshake->message_number < 32, S2N_ERR_SAFETY);
     return S2N_RESULT_OK;
 }
+
+S2N_RESULT s2n_handshake_set_finished_len(struct s2n_connection *conn, uint8_t len)
+{
+    RESULT_ENSURE_REF(conn);
+    RESULT_ENSURE_GT(len, 0);
+    RESULT_ENSURE_LTE(len, sizeof(conn->handshake.server_finished));
+    RESULT_ENSURE_LTE(len, sizeof(conn->handshake.client_finished));
+
+    uint8_t *finished_length = &conn->handshake.finished_len;
+    if (*finished_length == 0) {
+        *finished_length = len;
+    }
+
+    RESULT_ENSURE_EQ(*finished_length, len);
```

Comment:
What exactly is the point of this check? Why would this value change over the course of the handshake?

Reply:
I added a comment, see if that explains. Short version: it shouldn't change, that's why we're checking :)